multiSnake.py

```diff
@@ -0,0 +1,152 @@
+
+import pygame
+from pygame.locals import *
+from random import randint
+import time
+
+class multiSnake:
+    class snake:
+        def __init__(self, x, y, gameDisplay, blockSize):
+            self.coords=[(x,y)]
+            self.gridCoords = [(x/10, y/10)]
+            self.blockSize = blockSize
+            self.snakeBlock = pygame.image.load('snake.png')
+            self.alive = True
+            self.x_change = 0
+            self.y_change = 1
+            self.growX = False
+            self.growY = False
+        def kill(self):
+            self.alive = False
+            self.coords = []
+            self.gridCoords = []
+    def __init__(self):
+        pygame.init()
+        self.black = (0,0,0)
+        self.white = (255,255,255)
+        self.red = (255,0,0)
+        pygame.display.set_caption('MultiSnake')
+        self.clock = pygame.time.Clock()
+        self.spaceTaken = {}
+        pygame.joystick.init()
+        joystick_count = pygame.joystick.get_count()
+        for i in range(joystick_count):
+            joystick = pygame.joystick.Joystick(i)
+            joystick.init()
+            buttons = joystick.get_numbuttons()
+            for i in range(buttons):
+                button = joystick.get_button(i)
+
+
+        for i in range(int(self.rect.width/10)):
+            self.spaceTaken[i] = {}
+        self.players = []
+        self.playerCount = 2
+        
+        self.x = (self.rect.width * 0.2)/10
+        self.y = (self.rect.height * 0.2)/10
+        for i in range(self.playerCount):
+            if (i == 0):
+                self.players.append(self.snake(self.rect.width * 0.2, self.rect.height * 0.2, self.display, 10))
+            elif (i == 1):
+                self.players.append(self.snake(self.rect.width * 0.8, self.rect.height * 0.2, self.display, 10))
+            elif (i == 2):
+                self.players.append(self.snake(self.rect.width * 0.2, self.rect.height * 0.8, self.display, 10))
+            elif (i == 3):
+                self.players.append(self.snake(self.rect.width * 0.8, self.rect.height * 0.8, self.display, 10))
+        
+
+        self.game_loop()
+        pygame.quit()
+        quit()
+
+    
+    def checkForCollision(self, x, y):
+        if (x in self.spaceTaken.keys()):
+            if (y in self.spaceTaken[x].keys()):
+                return True
+        return False
+    def onevent(self, event):
+        x_change = 0
+        y_change = 1
+        gameExit = False
+        while not gameExit:

```

Comment:
If you're using the plethora api, this while loop is causing the game to work time as hard as it should have to.

Reply:
Follow up, it seems like this might be the pre-plethora-api version of snake? Ideally the plethora implementations should be in the arcade directory rather than at the highest directory, I believe.